include/abb_librws/rws_rapid.h

```diff
@@ -371,6 +372,243 @@ struct RAPIDRecord : public RAPIDSymbolDataAbstract
+  {
+    if (this != &other)
+    {
+      if (array_type_name_ == other.array_type_name_)
+      {
+        if (container_.size() == other.container_.size())
+        {
+          for (size_t i = 0; i < container_.size(); ++i)
+          {
+            container_.at(i).parseString(other.container_.at(i).constructString());
+          }
+        }
+      }
+    }
+
+    return *this;
+  }
+
+  value_type& operator[](const size_t index) {
+    ensureIndexIsValid(index);
+    return container_[index];
+  }
+
+  value_type const& at(const size_t index) const {
+    ensureIndexIsValid(index);
+    return container_.at(index);
+  }
+
+private:
+  /**
+   * \brief A method to remove a character from a string and count the number of times it occurred.
+   *
+   * \param input for the string to search.
+   * \param character specifying the character to search for.
+   *
+   * \return unsigned int containing the number of times the character occurs.
+   */
+  unsigned int countCharInString(std::string input, const char character)
+  {
+    bool done = false;
+    unsigned int count = 0;
+    size_t position = 0;
+
+    do
+    {
+      position = input.find_first_of(character);
+
+      if (position != std::string::npos)
+      {
+        ++count;
+        input.erase(position, 1);
+      }
+      else
+      {
+        done = true;
+      }
+    } while (!done);
+
+    return count;
+  }
+
+  /**
+   * \brief A method to extract delimited substrings in a string.
+   *
+   * \param input containing the string with delimited substrings.
+   *
+   * \return std::vector<std::string> containing the extracted substrings.
+   */
+  std::vector<std::string> extractDelimitedSubstrings(const std::string& input)
+  {
+    // Prepare the input by removing any starting and ending '[' respective ']'
+    std::string temp_0(input);
+    size_t position_1 = 0;
+    size_t position_2 = 0;
+
+    position_1 = temp_0.find_first_of('[');
+    position_2 = temp_0.find_last_of(']');
+
+    if (position_1 != std::string::npos && position_2 != std::string::npos)
+    {
+      temp_0.erase(position_1, 1);
+      temp_0.erase(position_2 - 1, 1);
+    }
+
+    // Extract and merge the delimited substrings in the prepared input string.
+    std::stringstream ss(temp_0);
+    std::vector<std::string> values;
+    std::string temp_1;
+    std::string temp_2;
+    int counter = 0;
+
+    while (std::getline(ss, temp_1, ','))
+    {
+      if (!temp_1.empty())
+      {
+        counter += countCharInString(temp_1, '[');
+        counter -= countCharInString(temp_1, ']');
+
+        if (counter == 0)
+        {
+          if (!temp_2.empty())
+          {
+            values.push_back(temp_2 + temp_1);
+            temp_2 = "";
+          }
+          else
+          {
+            values.push_back(temp_1);
+          }
+        }
+        else
+        {
+          temp_2 += temp_1 + ",";
+        }
+      }
+    }
+
+    return values;
+  }
```

Comment:
These functions are copypasted from `RAPIDRecord`. Don't repeat yourself, extract them to standalone utility functions.